### PR TITLE
Feat: add data provider for workflows thru data env0_environment

### LINKED ISSUE #956: add data provider for workflows thru `data env0_environment` or new `data env0_workflow_environment`
**Is your feature request related to a problem? Please describe.**
My main goal is to be able to pull data on a workflow - specifically the sub-environment ids.
I need the sub-environment ids in order to assign variables to the sub-environment post-deployment.

**Describe the solution you'd like**
either `env0_environment` returns more data about the workflow 
or a specific data source for `env0_workflow_environment`  (no strong opinions)

**Describe alternatives you've considered**

**Additional context**

This is the API endpoint example:

```bash
curl --request GET \
     --url https://api.env0.com/environments/175f5d24-7d14-4177-95d2-6d5b42ed86e5 \
     --header 'accept: application/json' \
     --header 'authorization: Basic bHZ2eH...' \
     --header 'content-type: application/json'
```

and response:

```json
{
  "lockStatus": null,
  "nextScheduledDates": null,
  "id": "175f5d24-7d14-4177-95d2-6d5b42ed86e5",
  "createdAt": "2023-07-07T18:25:04.217Z",
  "updatedAt": "2024-09-19T17:46:27.162Z",
  "name": "Workflow - ACME - Prod",
  "organizationId": "bde19c6d-d0dc-4b11-a951-8f43fe49db92",
  "projectId": "6fd1b0fc-733f-4caf-aa38-052d01c00d3f",
  "userId": "google-oauth2|110563071352660869820",
  "workspaceName": "env0175f5d",
  "requiresApproval": true,
  "status": "ACTIVE",
  "driftStatus": "DRIFTED",
  "latestDeploymentLogId": "2be1e51b-4677-4c4b-b44b-d8abdfa24c42",
  "lifespanEndAt": null,
  "markedForAutoDestroy": 0,
  "isArchived": false,
  "continuousDeployment": true,
  "pullRequestPlanDeployments": true,
  "vcsPrCommentsEnabled": false,
  "vcsCommandsAlias": null,
  "autoDeployOnPathChangesOnly": true,
  "autoDeployByCustomGlob": null,
  "terragruntWorkingDirectory": "",
  "isSingleUseBlueprint": null,
  "workflowEnvironmentId": null,
  "isRemoteBackend": false,
  "k8sNamespace": null,
  "isRemoteApplyEnabled": null,
  "latestDeploymentLog": {
    "output": {},
    "error": null,
    "costEstimation": null,
    "customEnv0EnvironmentVariables": null,
    "plan": null,
    "planSummary": null,
    "workflowFile": {
      "environments": {
        "network": {
          "name": "Network",
          "templateName": "wf-a-network",
          "templateId": "8a295a95-86bb-44ee-9d97-e43cee016112",
          "templateType": "terraform",
          "environmentId": "12a5f563-712c-415d-95a3-1652606ce150",
          "requiresApproval": false
        },
        "db": {
          "name": "DB",
          "templateName": "wf-b2-database",
          "templateId": "a08ba26b-e77b-4808-87ed-022f1a7bb99e",
          "templateType": "pulumi",
          "environmentId": "22803c3a-8b63-425f-b99a-5a36a634cc28",
          "requiresApproval": false
        },
        "compute": {
          "name": "Compute",
          "templateName": "wf-b1-compute",
          "needs": [
            "network"
          ],
          "templateId": "f4cc4f7c-0a94-4173-968b-3e84af0fb367",
          "templateType": "terraform",
          "environmentId": "46f1df2c-1cfe-4200-b80e-cce0f6b9bc6b",
          "requiresApproval": false
        },
        "billing_service": {
          "name": "Billing Service",
          "templateName": "wf-c-service",
          "disabled": false,
          "needs": [
            "compute",
            "db"
          ],
          "templateId": "9abdb40f-38b6-4ac9-9bb8-a8785569399a",
          "templateType": "helm",
          "environmentId": "7f43cc95-b137-4148-bc67-24a8f671a93d",
          "requiresApproval": false
        },
        "metrics_service": {
          "name": "Matrix Service",
          "templateName": "wf-c-service",
          "disabled": false,
          "needs": [
            "compute"
          ],
          "templateId": "9abdb40f-38b6-4ac9-9bb8-a8785569399a",
          "templateType": "helm",
          "environmentId": "b4b1f15b-fc42-4ae7-bee8-fc391c7e026c",
          "requiresApproval": false
        },
        "db_monitoring_service": {
          "name": "DB Monitoring Service",
          "templateName": "wf-c-service",
          "disabled": false,
          "needs": [
            "db"
          ],
          "templateId": "9abdb40f-38b6-4ac9-9bb8-a8785569399a",
          "templateType": "helm",
          "environmentId": "75f9e975-2c48-4603-8130-9394932f1ea7",
          "requiresApproval": false
        }
      },
      "settings": {
        "environmentRemovalStrategy": "destroy"
      }
    },
    "id": "2be1e51b-4677-4c4b-b44b-d8abdfa24c42",
    "createdAt": "2024-09-12T18:39:43.312Z",
    "updatedAt": "2024-09-12T18:42:50.286Z",
    "type": "deploy",
    "startedBy": "google-oauth2|110563071352660869820",
    "queuedAt": "2024-09-12T18:39:43.312Z",
    "startedAt": "2024-09-12T18:39:49.000Z",
    "finishedAt": "2024-09-12T18:42:50.196Z",
    "reviewers": [],
    "status": "SUCCESS",
    "blueprintId": "408e7443-6bdf-49cb-9ad2-1080ebf32016",
    "blueprintName": "Workflow - ACME",
    "blueprintRepository": "https://github.com/env0/acme-demo",
    "blueprintRevision": "main",
    "blueprintPath": "workflows/",
    "gitUser": null,
    "targets": null,
    "gitAvatarUrl": null,
    "comment": null,
    "prNumber": null,
    "blueprintType": "workflow",
    "environmentId": "175f5d24-7d14-4177-95d2-6d5b42ed86e5",
    "resourceCount": 13,
    "isSkippedApply": null,
    "workflowDeploymentId": null,
    "workflowDeploymentOptions": null,
    "isScheduledRun": false,
    "abortedBy": null,
    "triggerName": "user",
    "driftDetected": null,
    "stateVersionId": null,
    "deploymentDurationMinutes": null
  },
  "user": {
    "email": "andrew.way@env0.com",
    "created_at": "2021-03-25T19:59:28.357Z",
    "given_name": "Andrew",
    "user_id": "google-oauth2|110563071352660869820",
    "name": "Andrew Way",
    "picture": "https://lh3.googleusercontent.com/a/ACg8ocJ1qa3WEUOrrSIQPuVZNHFmdBe7clVtsjQ9dwAGTQI72Ha9WLNE=s96-c",
    "family_name": "Way",
    "last_login": "2024-09-20T14:54:55.227Z",
    "app_metadata": {}
  },
  "isLocked": false
}
```

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
resolves #956 

### Solution

To keep the implementation simple I decided to use the existing data_environment. (Instead of creating a new data source).

1. Updated the schema.
2. Added the code the writes to the schema.
3. Updated the acceptance tests and the harness tests.

### CODE REVIEW

File: env0/data_environment.go [hunk 1]
```diff
@@ -98,6 +98,25 @@ func dataEnvironment() *schema.Resource {
...
 				Computed:    true,
 			},
+			"sub_environment_configuration": {
+				Type:        schema.TypeList,
+				Description: "the sub environments of the workflow environment. (Empty for non workflow environments)",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeString,
```
Comment (TomerHeber) [inline env0/data_environment.go:108]: for now I have added 'alias' and 'id'. More can be added in the future if required.

File: env0/data_environment_test.go [hunk 2]
```diff
@@ -32,6 +32,13 @@ func TestEnvironmentDataSource(t *testing.T) {
 			BlueprintId:       template.Id,
 			BlueprintRevision: "revision",
 			Output:            []byte(`{"a": "b"}`),
+			WorkflowFile: &client.WorkflowFile{
+				Environments: map[string]client.WorkflowSubEnvironment{
+					"db": {
```
[the remark this answers is not in the document]
Comment (TomerHeber) [reply env0/data_environment_test.go:37]: sure! added!

File: env0/data_environment.go [hunk 3]
```diff
@@ -98,6 +98,25 @@ func dataEnvironment() *schema.Resource {
 				Description: "The token id used for repo integrations (Used by Gitlab or Azure DevOps)",
 				Computed:    true,
 			},
+			"sub_environment_configuration": {
```
[the remark this answers is not in the document]
Comment (TomerHeber) [reply env0/data_environment.go:103]: I'm unsure I understand the question.
But I'll try to answer...

The environment_resource does return this... but is not the required implementation.

File: env0/data_environment_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (TomerHeber) [reply env0/data_environment_test.go:37]: I also just added a sort - too keep order consistent.

File: env0/data_environment.go [hunk 4]
```diff
@@ -127,6 +148,30 @@ func dataEnvironmentRead(ctx context.Context, d *schema.ResourceData, meta inter
...
+		for alias, subenv := range environment.LatestDeploymentLog.WorkflowFile.Environments {
+			subEnvironment := map[string]interface{}{
+				"id":    subenv.EnvironmentId,
+				"alias": alias,
+			}
+			subEnvironments = append(subEnvironments, subEnvironment)
+		}
+	}
+
+	slices.SortFunc(subEnvironments, func(a, b interface{}) int {
```
Comment (TomerHeber) [inline env0/data_environment.go:163]: the order in a map can be inconsistent...
Added a sort to the slice to be sure order is always the same.

Comment (TomerHeber) [reply env0/data_environment.go:163]: this is mostly for tests... but who knows... maybe this inconsistency can also cause real issues in the future...

**Review by GiliFaroEnv0 — APPROVED** [review]
NICE WORK AS ALWAYS

**Pull request merged**